Multisig/UI/Settings/OwnerKeyManagement/AddDelegateKeyController.swift

```diff
@@ -0,0 +1,225 @@
+            // 3.1. on success, create delegate on backend
+            case .success(let signatureData):
+
+                // 4. send message and signature to the backend
+                self.sendToBackend(
+                    delegateAddress: delegatePrivateKey.address,
+                    signature: signatureData
+                ) { [weak self] sendResult in
+                    guard let self = self else { return }
+
+                    switch sendResult {
+                    // 4.1. on success, store the delegate key association
+                    //             and the delegate key in the keychain
+                    case .success:
+                        // store the delegate key association to the ledger key in the database
+
+                        // get the ledger key info
+                        let keyInfo: KeyInfo
+                        do {
+                            keyInfo = try self.loadKeyInfo()
+                        } catch {
+                            self.abortProcess(error: error)
+                            return
+                        }
+
+                        // modify to set delegate address
+                        keyInfo.delegateAddressString = delegatePrivateKey.address.checksummed
+
+                        // store the delegate key in keychain
+                        do {
+                            // create or update the key in the keychain
+                            try delegatePrivateKey.save()
+
+                            // save the database modifications
+                            keyInfo.save()
+                        } catch {
+                            // at this point, we registered delegate in the backend, will ignore that because
+                            // the delegate can be overriden in the future attempts.
+
+                            // however, we rollback the database changes to the KeyInfo
+                            keyInfo.rollback()
+                        }
+
+                        // post notification so that UI state can be updated
+                        NotificationCenter.default.post(name: .ownerKeyUpdated, object: nil)
+
+                        // trigger push notification registration
+                        App.shared.notificationHandler.signingKeyUpdated()
+
+                        self.completeProcess()
+                        break
+
+                    // 4.2. on error - show to the user, abort, close/completion
+                    case .failure(let error):
+                        self.abortProcess(error: error)
+                        break
+                    }
+                }
+
+                break
+
+            // 3.2. on error, show erro to the user, abort&complete
+            case .failure(let error):
+                self.abortProcess(error: error)
+                break
+            }
+        }
+    }
+
+    func loadKeyInfo() throws -> KeyInfo {
+        // get the ledger key info
```

Comment:
comments are outdated here and in other place in this file. Should be `get the key info`